### PR TITLE
feat(eslint-plugin): [require-array-sort-compare] add `ignoreStringArrays` option

### LINKED ISSUE #1963: [require-array-sort-compare] add option to ignore string arrays
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/require-array-sort-compare": []
  }
}
```

```TS
['c', 'b', 'a'].sort();
```

**Expected Result**
no error

**Actual Result**
`error: Require 'compare' argument. [@typescript-eslint/require-array-sort-compare]`

**Additional Info**
In the original rule implementation PR, there's a comment that an `ignoreStringArrays` option would be useful but can be punted on until someone asks for it: https://github.com/typescript-eslint/typescript-eslint/pull/261#issuecomment-463439800

I'm asking for it :)

The spirit of the rule makes sense but it's cumbersome to sprinkle comparator functions across lots of test files that are just sorting string arrays for the purposes of stable deep comparison. Why should we have to provide a comparator function for sorting a string array? The language handles this natively for most cases without the need to supply a custom function.

I would like to have the `ignoreStringArrays` option. And IMO it would be nice to set that to true by default, but I understand if for back-compat or for purposes of having devs think about edge cases like cross-locale comparison it makes sense to keep this option off by default.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.27.0` |
| `@typescript-eslint/parser`        | `2.27.0` |
| `TypeScript`                       | `3.6.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |

### PR DESCRIPTION
**Pull request by sosukesuzuki**
Fixes #1963

### CODE REVIEW

File: packages/eslint-plugin/src/rules/require-array-sort-compare.ts [hunk 1]
```diff
@@ -17,13 +31,35 @@ export default util.createRule({
...
+     * @param node
+     */
+    function isStringArrayNode(node: TSESTree.ArrayExpression): boolean {
+      return node.elements.every(element => {
+        const type = checker.getTypeAtLocation(
+          service.esTreeNodeToTSNodeMap.get(element),
+        );
+        return util.getTypeName(checker, type) === 'string';
+      });
+    }
```
Comment (bradzacher) [inline packages/eslint-plugin/src/rules/require-array-sort-compare.ts:61]: this only works for array expressions that have `sort` directly called on them: `['a'].sort()`

A better solution would be to check the type of the `node` instead, and extract the type of the generic parameters.

All array types (`string[]` and `Array<string>`) are treated as `Array<string>` in code.
All tuple types (`[a, b, c]`) are treated as `Tuple<T1, T2, ...>` (a special variadic type).

```ts
const type = checker.getTypeAtLocation(
  service.esTreeNodeToTSNodeMap.get(node),
);
if (checker.isArrayType(type) || checker.isTupleType(type)) {
  const typeArgs = checker.getTypeArguments(type);
  return checker.getTypeArguments(type).every(
    arg => util.getTypeName(checker, arg) === 'string',
  );
}
return false;
```

File: packages/eslint-plugin/tests/rules/require-array-sort-compare.test.ts [hunk 2]
```diff
@@ -152,5 +176,56 @@ ruleTester.run('require-array-sort-compare', rule, {
...
+    {
+      code: `
+        const one = 1;
+        const two = 2;
+        const three = 3;
+        [one, two, three].sort();
+      `,
+      errors: [{ messageId: 'requireCompare' }],
+      options: [{ ignoreStringArrays: true }],
+    },
```
Comment (bradzacher) [inline packages/eslint-plugin/tests/rules/require-array-sort-compare.test.ts:229]: add tests for:
```ts
declare const x: string[];
x.sort();
```

**Comment by sosukesuzuki** [comment]
Thank you for your review. I fixed all reviewed points!

**Comment by finnigantime** [comment]
@sosukesuzuki Thank you for adding this!

**Pull request merged**